### DISCUSSION TITLE
Pictures preview

### DISCUSSION THREAD

**Discussion by ghost** [retypeapp/retype#186] [Ideas]
Hello, I was thinking of a preview function of the image on mouse over or in href but attached to the page.
I could notice that even if the image takes the whole space, on big definition it appears small and it's difficult to see correctly.
![Pbw1fYuWgF](https://user-images.githubusercontent.com/61574808/138743134-792aeb48-57fd-457b-88a0-76d6171e2235.gif)
And again thank you for retype, the more I use it the more I find it powerful, complete and simple at the same time.

**Comment by geoffreymcgill** [comment]
Hi @febreze21. Thanks for the feedback and we're happy to hear Retype is helping.

Can you provide some more details regarding your image preview suggestion? I'm not sure I 100% understand the functionality.

One suggestion would be to make the image linkable, so clicking would go directly to the image.

Another obvious solution would be making the image smaller.

**Reply by ghost** [reply]
"One suggestion would be to make the image linkable, so clicking would go directly to the image."
yes sure. 

"Another obvious solution would be making the image smaller."
yes.

I mean a thing like this:
https://www.w3schools.com/howto/howto_css_zoom_hover.asp

![IbWDT1Mqnr](https://user-images.githubusercontent.com/61574808/138759787-7f2d248d-43d8-4b77-9a2a-883d65ab85c7.gif)

A sort of zoom on mouse hover.

https://codepen.io/RyGanley/pen/JKwaC

https://user-images.githubusercontent.com/61574808/138760651-4a2621ea-5c32-4f5a-a6f5-e3bcda29cf75.mp4

To see the image in full screen and then return to the tutorial without leaving the page with a cross and arrows to navigate between images. This would be useful for tutorials with screenshots.


Maybe like a carousel.

**Comment by ghost** [comment]
i found another suggestion: Flickity ( awesome open source project for carousels).
[https://flickity.metafizzy.co/options.html#carousel-cell2](https://flickity.metafizzy.co/options.html#carousel-cell2)

**Comment by DannyBen** [comment]
I think the Retype team needs to tread lightly here.

More often than not, zooming / gallery photo effects have a limited "shelf life". Sites using the practices that were common 2 years ago, feel dated today.

In my opinion, these guidelines should be considered:

1. Nothing should happen to images, unless explicitly requested by the user (the one who designs the Retype site)
2. If implementing anything, it should be along the lines of the most minimal, least animated, least obtrusive solution. Personally, I find something like [this one](https://photoswipe.com/) pleasant, meaning: Click to zoom to a black modal, arrows move left and right, ESC or click on the black, closes.

Obviously, the solution needs to be responsive and mobile friendly, so in the above example both arrows and swipe do the same thing with smooth snapping.

As I see a big animation of "hover to zoom" in this discussion, I must say this feels like problematic UX design to me.

**Comment by fabriciomurta** [comment]
Maybe retype can handle it, whenever it needed to resize an image to something not its original size, then it could become "clickable" so it is displayed in a modal like when a listing picture is clicked in eBay. It'd be a problem though, if the image is a link. But at the same time, yes, having retype not fiddle with the figure more than it should -- unless instructed to do so -- should be preferred.

Besides, sort of a "workaround" when you want images to be full-size viewable would be to turn them into links to themselves, that is:

`[![](static/sample.jpg)](static/sample.jpg)`

So when you click the image, it will be opened in the browser alone, allowing for further browser-driven zooming and panning if needed. This should at least be more compatible in the long run, although perhaps a bit distracting for readers.

**Reply by ghost** [reply]
Yes sure.  @DannyBen said better than me what i mean. With https://photoswipe.com/ 
looking really great. But yes href the image with her own directory is fine.

**Comment by ghost** [comment]
As @DannyBen said, to rephrase my request, I think implementing https://photoswipe.com/ or something similar would add a real plus to retype.

**Comment by ghost** [comment]
What do you think about photoswipe in retype @geoffreymcgill ?

Personally I am doing a windows server documentation. Because of the graphical interface I have to make a lot of screenshots. I think that for a reader, it would be easier to navigate with the screenshots which follow each other in full screen.

![ePtvM8Lhrf](https://user-images.githubusercontent.com/61574808/140056206-72a9065a-87e6-408a-a3ea-3bc09b860a4d.gif)
